androidcomponents/androidcomponents.go

```diff
@@ -88,165 +97,165 @@ func Ensure(androidSdk *sdk.Model, gradlewPath string) error {
+func (i installer) getDependencyCases() map[string]func(match string) error {
+	return map[string]func(match string) error{
+		`(Observed package id 'ndk-bundle' in inconsistent location)`: i.ndkInconsistentLocation,
+		`(NDK not configured)`:                                        i.ndkNotConfigured,
+		`failed to find target with hash string 'android-(.*)'\s*`:    i.target,
+		`failed to find Build Tools revision ([0-9.]*)\s*`:            i.buildTool,
+		`Could not find (com\.android\.support\..*)\.`:                i.extrasLib,
+	}
+}
 
-		if out, err := gradleCmd.RunAndReturnTrimmedCombinedOutput(); err != nil {
-			reader := strings.NewReader(out)
-			scanner := bufio.NewScanner(reader)
+func getDependenciesOutput(projectLocation string) (string, error) {
+	gradleCmd := command.New("./gradlew", "dependencies")
+	gradleCmd.SetStdin(strings.NewReader("y"))
+	gradleCmd.SetDir(projectLocation)
+	return gradleCmd.RunAndReturnTrimmedCombinedOutput()
+}
 
-			missingSDKComponentFound := false
+func (i installer) scanDependencies(foundMatches ...string) error {
+	out, err := getDependenciesOutput(filepath.Dir(i.gradlewPath))
+	if err == nil {
+		return nil
+	}
+	scanner := bufio.NewScanner(strings.NewReader(out))
+	for scanner.Scan() {
+		line := scanner.Text()
+		for pattern, callback := range i.getDependencyCases() {
+			re := regexp.MustCompile(pattern)
+			if matches := re.FindStringSubmatch(line); len(matches) == 2 {
+				if sliceutil.IsStringInSlice(matches[1], foundMatches) {
+					return fmt.Errorf("unable to solve a dependency installation for the output:\n%s", out)
+				}
+				if err := callback(matches[1]); err != nil {
+					fmt.Println(out)
+					return err
+				}
+				return i.scanDependencies(append(foundMatches, matches[1])...)
+			}
+		}
+	}
+	return nil
+}
+
+func (i installer) ndkNotConfigured(_ string) error {
+	log.Warnf("NDK not configured")
+
+	ndkComponent := sdkcomponent.SDKTool{SDKStylePath: "ndk-bundle", LegacySDKStylePath: "ndk-bundle"}
+	cmd := i.sdkManager.InstallCommand(ndkComponent)
+	cmd.SetStdin(strings.NewReader("y"))
 
-			for scanner.Scan() {
-				line := scanner.Text()
-				{
-					// failed to find target with hash string 'android-22'
-					targetPattern := `failed to find target with hash string 'android-(?P<version>.*)'\s*`
-					targetRe := regexp.MustCompile(targetPattern)
-					if matches := targetRe.FindStringSubmatch(line); len(matches) == 2 {
-						missingSDKComponentFound = true
+	log.Printf("Install and configure NDK bundle using:")
+	log.Printf("$ %s", cmd.PrintableCommandArgs())
+
+	if out, err := cmd.RunAndReturnTrimmedCombinedOutput(); err != nil {
+		return fmt.Errorf("output: %s, error: %s", out, err)
+	}
 
-						targetVersion := "android-" + matches[1]
+	bundlePath := filepath.Join(i.androidSDK.GetAndroidHome(), "ndk-bundle")
 
-						log.Warnf("Missing platform version found: %s", targetVersion)
+	log.Printf("Setting environment variable (%s) to:  %s", "ANDROID_NDK_HOME", bundlePath)
```

Comment:
log.Printf("Setting environment variable (ANDROID_NDK_HOME) to:  %s", bundlePath)